cpp/src/arrow/compute/exec.h

```diff
@@ -174,7 +175,10 @@ struct ARROW_EXPORT ExecBatch {
 
   explicit ExecBatch(const RecordBatch& batch);
 
-  static Result<ExecBatch> Make(std::vector<Datum> values);
+  /// \brief Infer the ExecBatch length from values.
+  static Result<int64_t> InferLength(const std::vector<Datum>& values);
```

Comment:
This would be the docs for the internal (.cc-only) `DoInferLength`, this one returns `Status` (wrapped in `Result`) for errors. I will rebase to fix conflicts and add these docs to `DoInferLength` instead.

Reply:
Yes, you're correct, this suggestion is in the wrong place.  Given it's internal you could probably use `//` and be more concise as well.  At a glance it wasn't obvious to me that the operation could "fail" and a small comment to explain those scenarios would be good.

Reply:
Just looked at what you added and it looks good